wagtail/snippets/views/snippets.py

```diff
@@ -360,94 +289,65 @@ def get_form_kwargs(self):
-        instance = form.instance
 
         context.update(
             {
                 "model_opts": self.model._meta,
                 "edit_handler": edit_handler,
                 "action_menu": action_menu,
-                "locale": None,
-                "translations": [],
                 "media": edit_handler.media + form.media + action_menu.media,
             }
         )
 
-        if self.locale:
-            context.update(
-                {
-                    "locale": instance.locale,
-                    "translations": [
-                        {
-                            "locale": locale,
-                            "url": reverse(
-                                "wagtailsnippets:add",
-                                args=[self.app_label, self.model_name],
-                            )
-                            + "?locale="
-                            + locale.language_code,
-                        }
-                        for locale in Locale.objects.all().exclude(
-                            id=instance.locale.id
-                        )
-                    ],
-                }
-            )
-
         return context
 
-    def form_valid(self, form):
-        response = super().form_valid(form)
-
-        hooks_result = self._run_after_hooks()
-        if hooks_result is not None:
-            return hooks_result
-
-        return response
-
 
 class Edit(EditView):
     template_name = "wagtailsnippets/snippets/edit.html"
     error_message = _("The snippet could not be saved due to errors.")
 
-    def _run_before_hooks(self):
-        for fn in hooks.get_hooks("before_edit_snippet"):
-            result = fn(self.request, self.object)
-            if hasattr(result, "status_code"):
-                return result
-        return None
+    def run_before_hook(self):
+        return self.run_hook("before_edit_snippet", self.request, self.object)
 
-    def _run_after_hooks(self):
-        for fn in hooks.get_hooks("after_edit_snippet"):
-            result = fn(self.request, self.object)
-            if hasattr(result, "status_code"):
-                return result
-        return None
+    def run_after_hook(self):
+        return self.run_hook("after_edit_snippet", self.request, self.object)
 
     def setup(self, request, *args, app_label, model_name, pk, **kwargs):
-        super().setup(request, *args, **kwargs)
-
         self.app_label = app_label
         self.model_name = model_name
         self.pk = pk
         self.model = self._get_model()
         self.edit_handler = self._get_edit_handler()
         self.object = self.get_object()
-        self.locale = self._get_locale()
+        super().setup(request, *args, **kwargs)
 
     def _get_model(self):
         return get_snippet_model_from_url_params(self.app_label, self.model_name)
 
     def _get_edit_handler(self):
         return get_snippet_edit_handler(self.model)
 
-    def _get_locale(self):
+    def get_locale(self):
```

Comment:
Should this logic (i.e. retrieving the locale from `self.object.locale`) be included in `LocaleMixin`? If so, which one takes precedence (compared to `?locale=` query param)?